gui/src/3D/controls/slider3D.ts

```diff
@@ -0,0 +1,220 @@
+import { Nullable } from "babylonjs/types";
+import { Observable } from "babylonjs/Misc/observable";
+import { Vector3 } from "babylonjs/Maths/math.vector";
+import { TransformNode } from "babylonjs/Meshes/transformNode";
+import { Scene } from "babylonjs/scene";
+import { Mesh } from "babylonjs/Meshes/mesh";
+import { Control3D } from "./control3D";
+import { MeshBuilder } from "babylonjs/Meshes/meshBuilder";
+import { PointerDragBehavior } from "babylonjs/Behaviors/Meshes/pointerDragBehavior";
+import { StandardMaterial } from "babylonjs/Materials/standardMaterial";
+import { Color3 } from "babylonjs/Maths/math.color";
+import { AbstractMesh } from "babylonjs/Meshes/abstractMesh";
+
+const SLIDER_MIN: number = 0;
+const SLIDER_MAX: number = 100;
+const SLIDER_VAL: number = 50;
+const SLIDER_STEP: number = 0;
+const SLIDER_SCALING: number = 2.0;
+
+/**
+ * Class used to create a slider in 3D
+ */
+export class Slider3D extends Control3D {
+    private _sliderBarMaterial: StandardMaterial;
+    private _sliderThumbMaterial: StandardMaterial;
+    private _sliderThumb: Mesh;
+    private _sliderBar: Mesh;
+
+    private _minimum: number;
+    private _maximum: number;
+    private _value: number;
+    private _step: number;
+
+    /** Observable raised when the sldier value changes */
+    public onValueChangedObservable = new Observable<number>();
+
+    /**
+     * Creates a new slider
+     * @param name defines the control name
+     */
+    constructor(name?: string) {
+        super(name);
+
+        this._minimum = SLIDER_MIN;

```

Comment:
Would it be possible to pass those values in the constructor, defaulting to those values? In the current implementation it will take 5 different calls if I want to change all values.

Reply:
To be consistent with the other GUI control, we decided not to pass these parameters in the constructor. The initial PR uses constructor parameters, but there was a code review feedback to remove it for the reason I mentioned earlier.